### PR TITLE
Stateful refactor

### PR DESCRIPTION
**Pull request by jake-shasteen**
Paired with @xuejinglu for this feature.

Ran into a rabbit hole with a Map component bug -- the Map currently creates a new Map every time it renders, making it impossible to both set the current event and maintain a popup. Also this activity of creating a new Map creates a jarring white flash whenever the map changes.

For this feature, we moved the state of most things into App, and added callbacks through prop propogation.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -67,11 +85,11 @@ export class App extends Component {
         if (data) {
           const eventList = data;
           this.setState({ events: eventList });
-          this.searchYouTube(eventList[0].title, this.handleVideoChange.bind(this));
-          this.setState({
-            lat: eventList[Math.floor(eventList.length / 2)].latitude,
-            lng: eventList[Math.floor(eventList.length / 2)].longitude,
-          });
+          this.searchYouTube(eventList[0].title, this.changeVideo.bind(this));
+          // this.setState({
```
Comment (jake-shasteen) [inline src/App.js]: Delete these comments

File: src/components/Map.js [hunk 2]
```diff
@@ -84,6 +91,8 @@ class Map extends Component {
       google.maps.event.addListener(marker, 'click', function (){ //eslint-disable-line
         if (currentSelectedMarker) {
           currentSelectedMarker.message.close();
+        }  else {
+          context.handleClick( pin.event );
```
Comment (kathy-ems) [inline src/components/Map.js]: Can you call context.props.changeCurrEvent( pin.event ) directly to cut out a few lines of code?

File: src/App.js [hunk 3]
```diff
@@ -11,29 +11,47 @@ import getloc from './utils/getloc';
...
-      // call getQuery on loc and date
-      this.getQuery(currlocation, today, today, 'music');
+      this.getQuery(currlocation, this.state.startDate, this.state.endDate, 'music');
+    });
+  }
+
+  changeLatLng(lat, lng) {
+    this.setState({
+      lat,
+      lng,
```
Comment (kathy-ems) [inline src/App.js:36]: When I was talking to Zack about react, he said that even though we can create new states throughout the project, we should instantiate them in the constructor with a null or empty string so there isn't confusion.

**Comment by kathy-ems** [comment]
@xuejinglu @jake-shasteen Great work on this! Very clean code that you added. A few comments added in-line. Otherwise, I have reviewed the code and tested it on my machine. The map pins are clickable and the map refocuses. The side bars are clickable and changes the video based on the clicked item.

Search is not functioning in this PR, which is expected because it was broken in master and we have another PR open that will resolve that issue.

**Comment by jake-shasteen** [comment]
Added commits to address issues @kathy-ems pointed out in inline comments

**Pull request merged**